pkg/controller/elasticloadbalancing/elb/controller.go

```diff
@@ -0,0 +1,394 @@
+	}
+
+	// in a successful response, there should be one and only one object
+	if len(response.LoadBalancerDescriptions) != 1 {
+		return managed.ExternalObservation{}, errors.New(errMultipleItems)
+	}
+
+	observed := response.LoadBalancerDescriptions[0]
+
+	tagsResponse, err := e.client.DescribeTagsRequest(&awselb.DescribeTagsInput{
+		LoadBalancerNames: []string{meta.GetExternalName(cr)},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(elb.IsELBNotFound, err), errDescribeTags)
+	}
+
+	// update the CRD spec for any new values from provider
+	current := cr.Spec.ForProvider.DeepCopy()
+	elb.LateInitializeELB(&cr.Spec.ForProvider, &observed, tagsResponse.TagDescriptions[0].Tags)
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errSpecUpdate)
+		}
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Available())
+
+	cr.Status.AtProvider = elb.GenerateELBObservation(observed)
+
+	upToDate, err := elb.IsUpToDate(cr.Spec.ForProvider, observed, tagsResponse.TagDescriptions[0].Tags)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errUpdate)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: upToDate,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.ELB)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	_, err := e.client.CreateLoadBalancerRequest(elb.GenerateCreateELBInput(meta.GetExternalName(cr),
+		cr.Spec.ForProvider)).Send(ctx)
+
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) { // // nolint:gocyclo
+	cr, ok := mgd.(*v1alpha1.ELB)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	response, err := e.client.DescribeLoadBalancersRequest(&awselb.DescribeLoadBalancersInput{
+		LoadBalancerNames: []string{meta.GetExternalName(cr)},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(elb.IsELBNotFound, err), errUpdate)
+	}
+
+	if len(response.LoadBalancerDescriptions) != 1 {
+		return managed.ExternalUpdate{}, errors.New(errMultipleItems)
+	}
+
+	observed := response.LoadBalancerDescriptions[0]
+
+	tagsResponse, err := e.client.DescribeTagsRequest(&awselb.DescribeTagsInput{
+		LoadBalancerNames: []string{meta.GetExternalName(cr)},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(elb.IsELBNotFound, err), errDescribeTags)
+	}
+
+	patch, err := elb.CreatePatch(observed, cr.Spec.ForProvider, tagsResponse.TagDescriptions[0].Tags)
```

Comment:
So `CreatePatch` here is being used to figure out what fields of the ELB need an update. This is done because there's no single API from AWS SDK to modify an ELB. (Added a comment  in the code as well)

Let me know if I understood your comment and answered it correctly.